compiler/src/compiler/cst_to_ast.rs

```diff
@@ -233,12 +232,12 @@ impl LoweringContext {
                     };
                 }
                 let receiver = match &name.kind {
-                    CstKind::Identifier(identifier) => Some(CallReceiver::Identifier(
-                        self.create_string(name.id.to_owned(), identifier.to_owned()),
-                    )),
-                    CstKind::StructAccess { struct_, dot, key } => Some(
-                        CallReceiver::StructAccess(self.lower_struct_access(struct_, dot, key)),
-                    ),
+                    CstKind::Identifier(identifier) => {
+                        Some(self.lower_identifier(name.id, identifier.to_string()))
+                    }
+                    CstKind::StructAccess { struct_, dot, key } => {
+                        Some(self.lower_struct_access(name.id.to_owned(), struct_, dot, key))
+                    }
                     _ => None,
                 };
```

Comment:
Doesn't the same reasoning that motivated the removal of a `CallReceiver` also apply here? i.e. not special-casing based on the type of the receiver and just doing `let receiver = self.lower_cst(name);` That would also allow for `lower_identifier` and `lower_struct_access` to be inlined.

Sidenote: It probably also makes sense to rename `CstKind::Call`'s `name` to `receiver`.